wordclock/wificfg.c

```diff
@@ -201,23 +195,34 @@ static void wifi_monitor_task(void *pvParameters) {
     }
     
     http_server_start(); //socket is reused for soft AP and normal mode
-    
-    wifiState_t state = INVALID;
+    uint8_t prvStatus = STATION_IDLE;
+
     while (true) {
-        if (sdk_wifi_station_get_connect_status() != STATION_GOT_IP) {
-            if (state != SOFT_AP) { //only do something on a state change
-                printf("Woordclock is unable to connect to Wifi\n");
+        uint8_t status = sdk_wifi_station_get_connect_status();
+        if (status == prvStatus) {
+            SleepNI(10 * 1000); // Do nothing, wait for state change
+            continue;
+        }
+        printf("Connect status changed from: %d, to: %d\n", prvStatus, status);
+        switch (status) {
+            case STATION_IDLE:
+                wificfg_start_softAP();
+            break;
+            case STATION_CONNECTING:
+            break;
+            case STATION_WRONG_PASSWORD:
+                wificfg_start_softAP();
+            break;
+            case STATION_NO_AP_FOUND:
                 wificfg_start_softAP();
-                state = SOFT_AP;
-            }
-        } else {
-            if (state != CONNECTED) { 
-                printf("Woordclock is now connected to the Wifi\n");
+            break;
+            case STATION_CONNECT_FAIL:
+            break;
+            case STATION_GOT_IP:
                 wificfg_stop_soft_AP();
```

Comment:
Ik heb het idee dat het nog niet goed gaat.  Als ik mijn WIFI router restart, is de kans groot dat daarna de tijd niet meer synchroniseerd. Rood LED rechts boven.
Ik heb onderstaande hier toegevoegd, dan lijkt het goed te gaan

```
                for (;;){
                    SleepNI(1000);
                };

```

Reply:
Daarmee stop je dus eigenlijk de WiFicfg task, wat kan maar het voelt wel als een hack,
wat ik me goed kan voorstellen is dat de NTP client (en misschien andere netwerk gerelateerde dingen) er niet tegen kan dat een verbinding weg valt.
of dat het switchen tussen wel of niet softAP niet goed gaat naar de NTP taks